### DISCUSSION TITLE
Help on service lookup editor

### DISCUSSION THREAD

**Discussion by Rohanb1976** [serenity-is/Serenity#6412] [Help]
we have UN Location data in SQL server consisting of more then 1 Lacs records.
This data has to be loaded on diffrent fields (e.g. 4 fields
Origin, Destination, Location of Loading, Location of Unloading)
 in the dialog and we are using Service look up editor.

During the retrive call, there are multiple (4) list call triggered. This increases the network activity and makes the system slow. On the database side the indexes on the search fields are already added.

Can we trigger the 4 list calls in one call and then set the data to UI?

**Comment by wirble** [comment]
So you have 4 different tables, each one containing Origin, Destination, Loading and Unloading?  So on Serenity side, you have 4 different Repository/Row for each corresponding data...I am assuming.

Low tech method:
One option is maybe create a view on Sql server side with 4 different tables to create in the one view.  Use sergen to create data for the view.  Then use the servicelookup on the view as you normally would....

Not so Low tech method:
Second option is to create a custom controller that entends ServiceEndpoint and create a function that query both tables and return the data you need.  This custom service can then be called on the xxxdialog.ts side...

If anyone have other ideas, I would like to hear it also...

**Reply by Rohanb1976** [reply]
Hi.
There are no diffrent tables. There is only one table but 4 diffrent components (Select2). So it quires the table four times in the retrive call. Can we reduce those 4 calls to one?

**Comment by minhhungit** [comment]
@Rohanb1976 I guess you can create your new service endpoint, the service will call sql query with JOIN to aggegrate data, then pass service url for the servcie lookup editor

**Reply by Rohanb1976** [reply]
I tried to create a new editor extending servicelookup. In that i pass one parameter as LoadData. If the value of load data is false then it will not make a list call. But i get one error as service  parameter is required when the LoadData is false. So to overcome that i created a dummy list call. Question is how can i avaoid the dummylist call?

**Comment by stixoffire** [comment]
Call it only once , assign to a value in your dialog.

`protected MyLocationData: Array<isWhateverRowData>;`

in Constructor 
```
{
MyLocationData = new  Array<isWhateverRowData>();

MyLocationData  = this.dataretrieved for field Origin, 

//Assign that data to each of your other fields.
this.Destination = MyLocationData  ;
this.Location of Loading = MyLocationData  ;
this.Location of Unloading = = MyLocationData  ;
}
```
